Cygwin/X: In Xinerama-o-vision!

Layout screens according to their native window positions in Xinerama mode

---
 xserver/hw/xwin/winscrinit.c |   10 ++++++++++
 1 files changed, 10 insertions(+)

Index: xorg-server-1.5.3/xserver/hw/xwin/winscrinit.c
===================================================================
--- xorg-server-1.5.3.orig/xserver/hw/xwin/winscrinit.c	2009-02-06 17:30:09.184250000 +0000
+++ xorg-server-1.5.3/xserver/hw/xwin/winscrinit.c	2009-02-06 17:30:49.953125000 +0000
@@ -237,6 +237,16 @@
   else
     winErrorFVerb(2, "winScreenInit - Using software cursor\n");  
 
+  /*
+     Note the screen origin in a normalized coordinate space where (0,0) is at the top left
+     of the native virtual desktop area
+  */
+  dixScreenOrigins[index].x = pScreenInfo->dwInitialX - GetSystemMetrics(SM_XVIRTUALSCREEN);
+  dixScreenOrigins[index].y = pScreenInfo->dwInitialY - GetSystemMetrics(SM_YVIRTUALSCREEN);
+
+  ErrorF("Screen %d added at XINERAMA coordinate (%d,%d).\n",
+         index, dixScreenOrigins[index].x, dixScreenOrigins[index].y);
+
 #if CYGDEBUG || YES
   winDebug ("winScreenInit - returning\n");
 #endif
